src/ir/possible-contents.h

```diff
@@ -174,6 +190,37 @@ class PossibleContents {
     }
   }
 
+  // Returns cone type info. This can be called on non-cone types as well, and
+  // it returns a cone that best describes them. That is, this is like getType()
+  // but it also provides an indication about the depth, if relevant. (If cone
+  // info is not relevant, like when getType() returns none or unreachable, the
+  // depth is set to 0.)
+  ConeType getCone() const {
+    if (auto* literal = std::get_if<Literal>(&value)) {
+      return ExactType(literal->type);
+    } else if (auto* global = std::get_if<GlobalInfo>(&value)) {
+      return FullConeType(global->type);
+    } else if (auto* coneType = std::get_if<ConeType>(&value)) {
+      return *coneType;
+    } else if (std::get_if<None>(&value)) {
+      return ExactType(Type::unreachable);
+    } else if (std::get_if<Many>(&value)) {
+      return ExactType(Type::none);
+    } else {
+      WASM_UNREACHABLE("bad value");
+    }
+  }
+
+  // Returns whether the relevant cone for this, as computed by getCone(), is of
+  // full size, that is, includes all subtypes.
+  bool hasFullCone() const { return getCone().depth == FullDepth; }
```

Comment:
WDYT about using "UnboundedCone" or "InfiniteCone" instead of "FullCone" as the terminology? "Full" makes me think of something finite, when the whole point is that it is not finite.

Reply:
Hmm... to me a partial cone is something finite, and a full cone is the fullest cone there can be, which is infinite. Does that not make sense, do you think?

Reply:
It does make sense, but I have to think about it slightly more than I would like to. I guess I'm still internally translating "cone type" to "bounded type" every time I see it, and now translating "full cone type" to "unbounded type" as well. Maybe "full" makes the most sense for the cone terminology.

Reply:
I did some searching but I can't seem to find better ideas for terminology here...

Reply:
No worries, I think "full" makes sense.

Reply:
Sounds good. 

@aheejin , any thoughts on this?